core/internal/server/features.go

```diff
@@ -0,0 +1,89 @@
+package server
+
+import (
+	"context"
+
+	"github.com/Khan/genqlient/graphql"
+	"github.com/wandb/wandb/core/internal/gql"
+	"github.com/wandb/wandb/core/internal/observability"
+	spb "github.com/wandb/wandb/core/pkg/service_go_proto"
+)
+
+// ServerFeatures is responsible for providing the capabilities of a server
+// Features and capabilities are retrieved from the server via a GraphQL query
+type ServerFeatures struct {
+	Features map[string]ServerFeature
+}
+
+type ServerFeature struct {
+	DefaultValue string
+	Description  string
+	Enabled      bool
+	Name         string
+}
+
+func NewServerFeatures(
+	graphqlClient graphql.Client,
+	ctx context.Context,
+) *ServerFeatures {
+	if graphqlClient == nil {
+		return &ServerFeatures{
+			Features: make(map[string]ServerFeature),
+		}
+	}
+
+	features := map[string]ServerFeature{}
+
+	// Query the server for the features provided by the server
+	resp, err := gql.ServerFeaturesQuery(
```

Comment:
like we discussed make the feature loader lazy, only on first request.

Reply:
Added lazy loading + mutex to prevent multiple graphql calls.